### DISCUSSION TITLE
Cache entry invalidate on hit?

### DISCUSSION THREAD

**Discussion by 0x161e-swei** [ChampSim/ChampSim#441] [Q&A]
On `src/cache.cc` [line 241](https://github.com/ChampSim/ChampSim/blob/feature/350-inclusivity/src/cache.cc#L241), I see that the block is sent for invalidation when `hit` is true.   
Could someone help me understand why this is needed?

**Comment by ngober** [comment]
You've linked a line from the `feature/350-inclusivity` branch. It looks like I was trying to implement a form of exclusivity. You should look at the discussion in #350 for more information on what we are working on in that branch.

cc: @GinoAC

**Reply by 0x161e-swei** [reply]
I was pointed to this branch from discussion in #350. Is this branch still work-in-progress?  

Shouldn't exclusiveness/inclusiveness should be checked before invalidating requests are sent?
Also, by definition, a tag hit in an exclusive cache already means that upper levels of cache dont have copies of this line?

**Reply by ngober** [reply]
Yes, that branch is still WIP, and not at all feature-complete. I know Gino is working on this, so I'll defer to him on the particulars.

Regarding a tag hit in an exclusive cache, it is true that upper levels don't have copies of the line, so invalidations requests probably don't need to be sent, but the invalidation should happen because the data is being sent to the upper level after the hit.

**Reply by 0x161e-swei** [reply]
Cool. Thanks for getting back to me!  Looking forward to Gino's response on how close we are to feature-complete.

**Reply by GinoAC** [reply]
Hey! I'll be adding the user-defined exclusive and non-inclusive models over the next few weeks.

**Comment by 0x161e-swei** [comment]
I was also wondering how the Champsim maintainers are thinking about inclusiveness for the micro-op cache (DIB in champsim term).  
Since DIB is [defined as an lru_table](https://github.com/ChampSim/ChampSim/blob/develop/inc/ooo_cpu.h#L109), there does not seem to be a channel for L1I and I-TLB to invalidate DIB entries on evictions from L1I and I-TLB.
